.github/workflows/macos.yml

```diff
@@ -29,3 +29,79 @@ jobs:
         env:
           MAX_WARNINGS: ${{ matrix.conf.max_warnings }}
         run:  python3 ./scripts/count-warnings.py build.log
+
+  build_macos_release:
+    name: Release build
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install C++ compiler and libraries
+        run:  brew install librsvg $(./scripts/list-build-dependencies.sh -m brew -c clang)
+      - name: Log environment
+        run:  ./scripts/log-env.sh
+      - name: Inject version string
+        run: |
+          set -x
+          export VERSION=$(git describe --abbrev=5)
+          sed -i -e "s/AC_INIT(dosbox,git)/AC_INIT(dosbox,$VERSION)/" configure.ac
+          echo ::set-env name=VERSION::$VERSION
+      - name: Build
+        env:
+          FLAGS: -O3 -DNDEBUG -pipe
+        run: |
+          set -x
+          ./autogen.sh
+          # FIXME temporarily disable everything aside SDL2 due to linking issues
+          ./configure \
+            --enable-sdl-static \
+            --disable-opus-cdda \
```

Comment:
For Opus, we could include a prior Step:
``` yaml
      - name: Build static Opus library
        run: |
          set -x
          cd contrib/static-opus
          gmake -j "$(sysctl -n hw.physicalcpu)"
 ```

And then add these two to the `Build` step's `env:`

``` yaml
        env:
          FLAGS: ...
          OPUSFILE_CFLAGS="-I${{ github.workspace }}/contrib/static-opus/include -I${{ github.workspace }}/contrib/static-opus/include/opus"
          OPUSFILE_LIBS="${{ github.workspace }}/contrib/static-opus/lib/libopusfile.a ${{ github.workspace }}/contrib/static-opus/lib/libogg.a ${{ github.workspace }}/contrib/static-opus/lib/libopus.a -lm"
```

Reply:
True, we could use it… but it means downloading and rebuilding opus, opusfile, and ogg libraries on every push. And we already have these libraries downloaded for static linking via brew.

I would prefer to come up with a systemic solution for static linking via our build system instead - we need to solve it also for our Linux snapshots (they have screenshots functionality disabled to avoid linking libpng dynamically due to Ubuntu dropping libpng12 and SteamOS missing libpng16).

On the other hand - we download tarballs with these libraries from github archives, so it's not such a big deal. I'll try it to see how much time CI spends on compilation.

Reply:
+55s to the build time of release build (still, our Windows build times are much longer so it doesn't matter that much). Download times were negligible.

I find it's a good compromise until we'll have the proper solution in place :) Implemented in 6bb0eee32d18a29fb35e1f822790198d05839edf.

Reply:
> we already have these libraries downloaded for static linking via brew.

Oh, I didn't know that! I guess `pkg-config` (or more specifically our autotools wrapper call into it) only gives back shared libraries? That's too bad we don't have an easy way to flip between shared and static (without probably a lot of our own legwork in configure.ac).

Glad this is a workable solution in the meantime!

Reply:
Correct; surprisingly, it seems like .pc files on Linux are not a magic solution for static linking either - they never return path to .a file (and I'm not sure if flags are correct either); at this point I'm not sure how to correctly manage static linking using pkg-config only (perhaps we should expect user to manually pass path to .a file instead…). Most answers on StackOverflow are quite rubbish and assume that project is being built for a single OS or expect users to write their own .pc files :(